causy/graph_model.py

```diff
@@ -0,0 +1,241 @@
+import logging
+from abc import ABC
+from typing import Optional, List, Dict
+
+import torch.multiprocessing as mp
+
+from causy.graph import Graph
+from causy.graph_utils import unpack_run
+from causy.interfaces import (
+    PipelineStepInterface,
+    TestResultAction,
+    LogicStepInterface,
+    BaseGraphInterface,
+    GraphModelInterface,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractGraphModel(GraphModelInterface, ABC):
+    """
+    The graph model is the main class of causy. It is responsible for creating a graph from data and executing the pipeline_steps.
```

Comment:
I think you mean: The graph model is the main class of causy. It is responsible for initialising nodes for each variable that we have data from, then initializing a fully connected graph by adding edges between each pair of these nodes and then executing the pipeline_steps.